spyder/utils/ipython/tests/test_spyder_kernel.py

```diff
@@ -18,6 +17,23 @@
 from spyder.utils.ipython.kernelspec import SpyderKernelSpec
 
 
+def test_python_interpreter(tmpdir):
+    """Test the validation of the python interpreter."""
+    # Set a non existing python interpreter
+    interpreter = tmpdir.mkdir('interpreter')
```

Comment:
Let's change this to `interpreter = tmpdir.mkdir('interpreter').join("python")` to have a more realistic test.

Reply:
Also, you need to enclose the statement to the left in a `str` call to produce a string. So this really needs to be

    interpreter = str(tmpdir.mkdir('interpreter').join("python"))